src/lexers/PHPLexer.js

```diff
@@ -99,39 +121,45 @@ define(function (require, exports, module) {
                         default:
                             break;
                     }
+                    // reset modifier when identity term is parsed.
+                    modifier = null;
+                    isStatic = false;
                 }
             })
             // check if it's in function definition, turn on args mode.
             .addRule(/\(/, function () {
-                if (peek(state) === "function") {
-                    var ref = peek(results);
-                    if (!ref || ref.type !== "function") {
-                        ns.push(unnamedPlaceholder);
-                        results.push({
-                            type: "function",
-                            name: ns.join("::"),
-                            args: [],
-                            modifier: "unnamed",
-                            isStatic: false,
-                            line: line
-                        });
+                if (!literal && !comment) {
+                    if (peek(state) === "function") {
+                        var ref = peek(results);
+                        if (!ref || ref.type !== "function") {
+                            ns.push(unnamedPlaceholder);
+                            results.push({
+                                type: "function",
+                                name: ns.join("::"),
+                                args: [],
+                                modifier: "unnamed",
+                                isStatic: false,
```

Comment:
It doesn't need because this part of code just parsed the unnamed function in which modifiers are not supported.